docs/understanding-the-flutterflow-ui/projects/how-to-run-test-projects.md

```diff
@@ -0,0 +1,94 @@
+  - Firebase auth flow can't be tested. We always allow log in.
+  - API Calls can't be run or tested here.
+  - Refresh if animation actions are not working.
+  - Refresh if Clear TextFields actions are not working
+  - RevenueCat data is not loaded.
+  - Paywall actions execute as if the entitlement is active.
+  - Hero Animation may not work on dynamically generated widgets.
+  - Dropdown disabling does not work in
+  - Preview Mode.
+  - Tooltip does not work for some screen sizes in Preview Mode.
+</div>
+
+## Test Mode
+
+Test mode uses Flutter's "Hot Reload" feature, which lets you immediately see any changes made to code in an emulator or on-device. Running your app in _Test_ Mode helps you experiment, test UIs, and fix bugs faster.
+
+To run your app in Test Mode:
+
+1. Click on the **Test icon** or press **Cmd/Ctrl + I** (a keyboard shortcut). This will run your app in a new browser window.
+2. Switch to the app builder and make some changes, such as changing colors and alignment.
+3. Now switch to the Test Mode tab and click **Instant Reload** or press **Cmd/Ctrl + J** to see the changes in under 10 seconds.
+
+<h4>Test Mode Button Colors</h4>
+The test mode button color indicates the following status:
+  - Orange: Building
+  - Green: Ready
+  - Yellow: Expiring Soon
+  - Red: Expired
+<div class="alert alert--info" role="alert">
+#### When to Use Test Mode
+When designing and developing your app, Test Mode is the second most common testing mode (next to Local Run). It's Hot Reloading allows for fast iteration and testing. It's downsides compared to Local Run are its restricted time (30 min.) and lack of access to some features (e.g. camera).
+</div>
+<div class="alert alert--info" role="alert">
+<h4>Troubleshooting – Slow Loading</h4>
+If you see a progress bar at the top of the device running for over 15 seconds, try refreshing the page.
+</div>
+<div class="alert alert--warning" role="alert">
+#### Test Mode Limitations
+- 30 minute testing time. After the 30 minutes has expired, you can start another test mode.
+- If you see a grey "broken" screen with a sad face, it may be a DNS server issue with your network provider. We recommend using CloudFlare's 1.1.1.1 DNS server. Click here to see instructions.
+- Lottie animation may not load if you provide a variable path.
+- Cookies need to be enabled for Test Mode to function properly. They are only used for functional purposes.
+- If you see a progress bar where the phone outline should be that lasts longer than a 15 seconds, try refreshing the page.
+- The device screen can not be wider than the page's width.
+- Copy to Clipboard Action is not supported in Test Mode. Use Run Mode to avoid this issue.
+- Widgets having Shimmer or Tint animation might not appear properly.
+- Assets used within Custom Code might not appear properly.
+- Audio Recording actions do not work in Test Mode, use web publishing in Settings to test recording audio.
+</div>
+
+## Run Mode
```

Comment:
Also mention here somewhere that it will still run the web version of the project.